### PR TITLE
Load CANOpen dictionary from ESI file

### PR DESCRIPTION
**Pull request by leducp**

### CODE REVIEW

File: lib/include/kickcat/CoE/OD.h [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        Entry() = default;
+        ~Entry();
+
+        Entry(Entry const&) = delete;
+        Entry& operator=(Entry const& other) = delete;
+
+        Entry(Entry&& other);
+        Entry& operator=(Entry&& other);
+
+        uint8_t      subindex;
```
Comment (Rdk-T) [inline lib/include/kickcat/CoE/OD.h:166]: In table 48 "Get Entry Description Response" ETG1000.6 it specifies also the value info and datatype before bit len.
And no description.

There seems to be changes of order/mismatch with this table.

File: lib/src/CoE/EsiParser.cc [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        {
+            if (node_setting->GetText()[0] == '1') { flags |= Access::SETTING; }
+        }
+
+        return flags;
+    }
+
+    std::tuple<DataType, uint16_t> EsiParser::toType(XMLNode* node)
+    {
+        std::unordered_map<std::string, std::tuple<DataType, uint16_t>> BASIC_TYPES
```
Comment (Rdk-T) [inline lib/src/CoE/EsiParser.cc:192]: Why re declare the map at each call ?

File: lib/include/kickcat/CoE/OD.h (hunk 1 above)
Comment (leducp) [reply lib/include/kickcat/CoE/OD.h:166]: This entry is NOT the answer of the SDO, but an internal storage: it has to be serialized for such purpose.

Doing this way enable non POD management (so constructor/destructor, std::string or STL container, etc). It seems more important to have a good itnernal class than an already 'mappable' version for SDO Info service.

File: lib/src/CoE/EsiParser.cc (hunk 2 above)
Comment (leducp) [reply lib/src/CoE/EsiParser.cc:212]: Good catch :)

File: lib/CMakeLists.txt [hunk 3]
```diff
@@ -1,9 +1,16 @@
-set(KICKCAT_SOURCES 
+set(KICKCAT_SOURCES
   ${CMAKE_CURRENT_SOURCE_DIR}/src/Error.cc
   ${CMAKE_CURRENT_SOURCE_DIR}/src/protocol.cc
   ${CMAKE_CURRENT_SOURCE_DIR}/src/Time.cc
+
```
[the remark this answers is not in the document]
Comment (leducp) [reply lib/CMakeLists.txt:6]: voluntary to separate CoE section from the more general section.

**Review by Rdk-T — APPROVED** [review]

**Pull request merged**